compiler/accelerators/streamers/streamers.py

```diff
@@ -8,19 +10,83 @@ class StreamerType(StrEnum):
     Writer = "w"
 
 
+class StreamerFlag(StrEnum):
+    """
+    Enum that specifies special flags for a streamer dimension.
+
+    Attributes:
+    -----------
+    - Normal: 'n'
+      Indicates that no special flags apply.
+    - Irellevant : 'i'
+      Indicates a dimension is irrelevant for this operand.
+      In all cases, a zero should be set to the stride of this dimension.
+      For spatial dims, the resulting value is assigned to a "virtual streamer"
+      and will not be programmed.
+    - Reuse : 'r'
+      Only valid for temporal dims. Indicates that the values
+      will be reused temporally and should only be fetched once.
+      This results in the bound values being for this dimension fixed to 1.
+    """
+
+    Normal = "n"
+    Irrelevant = "i"
+    Reuse = "r"
+
+    def __bool__(self) -> bool:
+        """
+        Overrides the default boolean conversion to ensure that the 'None'
+        flag evaluates to False and all other flags evaluate to true
+        """
+        return self is not StreamerFlag.Normal
+
+
 class Streamer:
     """
     A representation of a single SNAX Streamer
     """
 
     type: StreamerType
-    temporal_dim: int
-    spatial_dim: int
+    temporal_dims: tuple[StreamerFlag, ...]
+    spatial_dims: tuple[StreamerFlag, ...]
 
-    def __init__(self, type: StreamerType, temporal_dim: int, spatial_dim: int) -> None:
+    def __init__(
+        self,
+        type: StreamerType,
+        temporal_dims: Sequence[StreamerFlag | Literal["n", "i", "r"]],
+        spatial_dims: Sequence[StreamerFlag | Literal["n", "i", "r"]],
+    ) -> None:
         self.type = type
-        self.temporal_dim = temporal_dim
-        self.spatial_dim = spatial_dim
+        temporal_dims = [
+            f if isinstance(f, StreamerFlag) else StreamerFlag(f) for f in temporal_dims
+        ]
+        self.temporal_dims = tuple(temporal_dims)
+        spatial_dims = [
+            f if isinstance(f, StreamerFlag) else StreamerFlag(f) for f in spatial_dims
+        ]
+        self.spatial_dims = tuple(spatial_dims)
+
+    @classmethod
+    def from_dim(
+        cls, type: StreamerType, temporal_dim: int, spatial_dim: int
+    ) -> "Streamer":
+        """
+        Returns a streamer with a specified temporal dim and spatial dim
+        as integers, without any flags set.
+        """
+        return cls(
+            type,
+            (StreamerFlag.Normal,) * temporal_dim,
+            (StreamerFlag.Normal,) * spatial_dim,
+        )
+
+    @property
+    def temporal_dim(self):
+        return len(self.temporal_dims)
+
+    @property
+    def spatial_dim(self):
+        return len(self.spatial_dims)
```

Comment:
The temporal and spatial dim for the streamers of a given operand is still valid.